src/marker.rs

```diff
@@ -680,6 +696,55 @@ impl MarkerExpression {
             (MarkerValue::QuotedString(l_string), MarkerOperator::Equal, MarkerValue::Extra) => {
                 extras.contains(l_string)
             }
+            (
+                MarkerValue::MarkerEnvVersion(MarkerValueVersion::PythonVersion),
+                operator,
+                MarkerValue::QuotedString(r_string),
+            ) => {
+                // ignore all errors block
+                (|| {
+                    // The right hand side is allowed to contain a star, e.g. `python_version == '3.*'`
+                    let (r_version, r_star) = Version::from_str_star(r_string).ok()?;
+                    let operator = operator.to_pep440_operator()?;
+                    // operator and right hand side make the specifier
+                    let specifier = VersionSpecifier::new(operator, r_version, r_star).ok()?;
+
+                    // We're passing a lot of versions here, e.g. 3.8 .. 3.100, but since we're
+                    // passing them in order it is expected that we either match early or not at all
+                    let compatible = python_versions
+                        .iter()
+                        .any(|l_version| specifier.contains(l_version));
+                    Some(compatible)
+                })()
+                .unwrap_or(true)
+            }
+            (
+                MarkerValue::QuotedString(l_string),
+                operator,
+                MarkerValue::MarkerEnvVersion(MarkerValueVersion::PythonVersion),
+            ) => {
+                // ignore all errors block
+                (|| {
```

Comment:
the error handling here is ugly but i unfortunately couldn't find a better to do this

Reply:
Yeah, what you want is the [`try_blocks` feature](https://doc.rust-lang.org/unstable-book/language-features/try-blocks.html). 

You could extract all of it into a function that returns `Result` rather than using a lambda but the JS IIFE like style is OK with me